Public/Src/Cache/ContentStore/Distributed/Sessions/DistributedContentSession.cs

```diff
@@ -165,5 +181,32 @@ private async Task<PutResult> RegisterPutAsync(Context context, CancellationToke
 
             return putResult;
         }
+
+        private async Task<BoolResult> RequestProactiveCopyIfNeededAsync(OperationContext context, ContentHash hash, UrgencyHint urgencyHint)
+        {
+            var hashArray = new[] { hash };
+
+            // First check in local location store, then global if failed.
+            var getLocationsResult = await ContentLocationStore.GetBulkAsync(context, hashArray, context.Token, UrgencyHint.Nominal, GetBulkOrigin.Local);

```

Comment:
Minor: Can we add a non-bulk ContentLocationStore.Get? Even if for now if is implemented by calling the bulk method, this keeps all the array allocations in one place that we can eliminate later when we add a real non-bulk path.

Reply:
While it would be useful, I think this is something we should do in a separate PR; This PR is starting to be too big and doing refactors might make it confusing.

Reply:
While it would be useful, I think this is something we should do in a separate PR; This PR is starting to be too big and doing refactors might make it confusing.

---
In reply to: [311186928](https://github.com/microsoft/BuildXL/pull/705#discussion_r311186928) [](ancestors = 311186928)